### PR TITLE
Created new formula for MockServer

### PR DESCRIPTION
**Pull request by jamesdbloom**
MockServer is a tool for mocking HTTP servers and for proxying HTTP & SSL.

### CODE REVIEW

**Comment by jamesdbloom** [comment]
Is there any update on when this pull request might be reviewed?  Thanks, James

File: Library/Formula/mockserver.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  version "3.9.15"
+  sha256 "0dbe6e78c3753c1da0381322ec0d348c2a3a02783fd65bf280a7bfaff02046d7"
+
+  depends_on :java => "1.6+"
+
+  def install
+    libexec.install Dir["*"]
+    bin.install_symlink "#{libexec}/bin/run_mockserver.sh" => "mockserver"
+
+    # add lib directory soft link
```
Comment (mikemcquaid) [inline Library/Formula/mockserver.rb]: Can kill this comment.

File: Library/Formula/mockserver.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    lib.install_symlink "#{libexec}/lib" => "mockserver"
+
+    # add log directory soft link
+    mockserver_log = var/"log"/"mockserver"
```
Comment (mikemcquaid) [inline Library/Formula/mockserver.rb]: `mockserver_log = var/"log/mockserver"

File: Library/Formula/mockserver.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  depends_on :java => "1.6+"
+
+  def install
+    libexec.install Dir["*"]
+    bin.install_symlink "#{libexec}/bin/run_mockserver.sh" => "mockserver"
+
+    # add lib directory soft link
+    lib.install_symlink "#{libexec}/lib" => "mockserver"
+
+    # add log directory soft link
```
Comment (mikemcquaid) [inline Library/Formula/mockserver.rb]: Can kill this comment.

File: Library/Formula/mockserver.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    server.close
+
+    # start MockServer
+    mockserver = fork do
+      exec "#{bin}/mockserver", "-serverPort", port.to_s
+    end
+
+    # check its started up correctly
+    loop do
+      `#{"curl -s \"http://localhost:" + port.to_s + "/status\" -X PUT"}`
```
Comment (mikemcquaid) [inline Library/Formula/mockserver.rb]: Use `Utils.popen` instead of backticks here.

File: Library/Formula/mockserver.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      break if $?.exitstatus == 0
+    end
+
+    # shut it down
+    system "curl", "-s", "http://localhost:" + port.to_s + "/stop", "-X", "PUT"
+
+    # wait for MockServer to stop
```
Comment (mikemcquaid) [inline Library/Formula/mockserver.rb]: Can kill all the comments, they are self-explanatory.

**Comment by jamesdbloom** [comment]
I have applied all suggested changes, is there any update on when this might be merged?  Thanks, James

**Comment by bfontaine** [comment]
Just need a :+1: from @Homebrew/owners since this is a new formula :wink:

**Comment by jamesdbloom** [comment]
Can you explain a little or point me at a website to explain how that process works and the rough timelines I might expect.

Thanks, James

**Comment by tdsmith** [comment]
Pulled; thanks for the PR and for your patience!

**Pull request closed without merging**